template.yaml

```diff
@@ -0,0 +1,67 @@
+Description: Upload an object to an S3 bucket, triggering a Lambda event, returning the object key as a Stack Output.
+Parameters:
+  Key:
+    Description: S3 Object key
+    Type: String
+    Default: test
+  Body:
+    Description: S3 Object body content
+    Type: String
+    Default: TEST CONTENT
+  abc1:
+    Description: S3 Bucket name (must not already exist)
+    Type: String
+Resources:
+  Bucket:
+    Type: AWS::S3::Bucket
+    DependsOn: BucketPermission
+    Properties:
+      BucketName: !Ref abc1
+      NotificationConfiguration:
+        LambdaConfigurations:
+        - Event: 's3:ObjectCreated:*'
+          Function: !GetAtt BucketWatcher.Arn
+  BucketPermission:
+    Type: AWS::Lambda::Permission
+    Properties:
+      Action: 'lambda:InvokeFunction'
+      FunctionName: !Ref BucketWatcher
+      Principal: s3.amazonaws.com
+      SourceAccount: !Ref "AWS::AccountId"
+      SourceArn: !Sub "arn:aws:s3:::${abc1}"
+  BucketWatcher:
+    Type: AWS::Lambda::Function
+    Properties:
+      Description: Sends a Wait Condition signal to Handle when invoked
+      Handler: index.handler
+      Role: !GetAtt LambdaExecutionRole.Arn
+      Code:
+        ZipFile: |

```

Comment:
As discussed, Upload lambda code as a zip in s3 and then use it here. We generally don´t use inline code in template as lamda codes are bigger and complex